modules/controlfreec/1.1/controlfreec.smk

```diff
@@ -233,86 +262,114 @@ rule _controlfreec_config:
+        BAF = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}.bam_minipileup.pileup.gz_BAF.txt"
+    conda: CFG["conda_envs"]["controlfreec"]
+    threads: CFG["threads"]["controlfreec_run"]
+    resources: **CFG["resources"]["controlfreec_run"]
+    wildcard_constraints:
+        normal_id="|".join(unmatched_normal_ids)
+    priority: 0
+    log:
+        stdout = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/run.stdout.log",
+        stderr = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/run.stderr.log"
+    shell:
+        "freec -conf {input.config} > {log.stdout} 2> {log.stderr} "
+
+
 rule _controlfreec_run:
     input:
-        config = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/config_WGS.txt",
-        tumour_bam = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}.bam_minipileup.pileup.gz",
-        normal_bam = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{normal_id}.bam_minipileup.pileup.gz",
+        config = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/config_WGS.txt",
+        tumour_bam = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}.bam_minipileup.pileup.gz",
+        normal_bam = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{normal_id}.bam_minipileup.pileup.gz",
     output:
-        info = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}.bam_minipileup.pileup.gz_info.txt",
-        ratio = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}.bam_minipileup.pileup.gz_ratio.txt",
-        CNV = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}.bam_minipileup.pileup.gz_CNVs",
-        BAF = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}.bam_minipileup.pileup.gz_BAF.txt"
+        info = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}.bam_minipileup.pileup.gz_info.txt",
+        ratio = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}.bam_minipileup.pileup.gz_ratio.txt",
+        CNV = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}.bam_minipileup.pileup.gz_CNVs",
+        BAF = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}.bam_minipileup.pileup.gz_BAF.txt"
     conda: CFG["conda_envs"]["controlfreec"]
     threads: CFG["threads"]["controlfreec_run"]
-    resources: mem_mb = CFG["mem_mb"]["controlfreec_run"]
+    resources: **CFG["resources"]["controlfreec_run"]
+    wildcard_constraints: 
```

Comment:
Is there anything different about the matched vs. unmatched rules? I don't think you actually need two rules, the output rule with the runs table should take care of matching the right matched vs. unmatched tumour/normal pairs.

Reply:
I was thinking it may be an issue if I had multiple runs that tries to read from the same unmatched normal at the same time, so I made a separate rule specifically for the unmatched group in case that ever needs to be configured, like set bam=1.

Reply:
I'm not too worried about it - control FREEC doesn't take long to run if the pileups are pre-generated, and you can just set a resource limit that prevents it from launching too many matched or unmatched jobs simultaneously. I think you would only need a separate unmatched normal rule if you were creating group jobs. Please do delete the redundant rule and the wildcard constraints.

Reply:
Okay, the redundant rule and wildcard constraint are removed.